src/bindings.rs

```diff
@@ -64,13 +84,23 @@ impl Preprocessor {
         let result = self.core.process(
             &src,
             Options {
-                filename: filename.map(|f| f.into())
-            }
+                filename: filename.map(|f| f.into()),
+            },
         );
 
         match result {
             Ok(output) => Ok(output),
             Err(err) => Err(as_javascript_error(err, self.core.source_map()).into()),
         }
     }
+
+    pub fn parse(&self, src: String, filename: Option<String>) -> Result<JsValue, JsValue> {
+        let result = self
+            .core
+            .parse(&src)
+            .map_err(|err| js_error("parse error".into()))?;
```

Comment:
TODO: surface the same kind of details as the process method if possible (it maybe fine to just actually call `process` upon encountering an error here)